core/services/document_service.py

```diff
@@ -91,7 +91,7 @@ async def query(
         query: str,
         auth: AuthContext,
         filters: Optional[Dict[str, Any]] = None,
-        k: int = 4,
+        k: int = 20,  # from contextual embedding paper
```

Comment:
should we keep this consistent in the retrieve functions also? Will require doc update